src/GShark/Geometry/Mesh.cs

```diff
@@ -548,6 +548,127 @@ private FaceData CountFaceEdges()
+            Line segment_AB = new Line(trianglePoints[0], trianglePoints[1]);
+            Line segment_BC = new Line(trianglePoints[1], trianglePoints[2]);
+            Line segment_CA = new Line(trianglePoints[2], trianglePoints[0]);
+            List<Line> segments = new List<Line>() { segment_AB, segment_BC, segment_CA };
+            List<Point3> edgesClosestPoints = new List<Point3>();
+            foreach (Line segment in segments)
+            {
+                Point3 segmentClosestPoint = segment.ClosestPoint(point);
+                edgesClosestPoints.Add(segmentClosestPoint);
+            }
+            return point.CloudClosestPoint(edgesClosestPoints);
+        }
+
+        public Point3 ClosestPoint(Point3 point)
+        {
+            List<MeshVertex> meshVertices = Vertices;
+
+            //All faces check 
+            System.Collections.Generic.IEnumerable<MeshFace> AllFaces = this.Faces;
+
+            var CloserPointToFace_List = new List<Point3>();
+
+            foreach (MeshFace face in AllFaces)
+            {
+                // Retrives the Vertices associated with each face
+                List<MeshVertex> GSVertices = face.AdjacentVertices();
+                // Converts from MeshVertex to Point3
+                List<Point3> FacePoints = GSVertices.ConvertAll(v => (Point3)v);
+
+                if (GSVertices.Count == 3)
+                {
+                    var trianglePoints = new Point3[] { FacePoints[0], FacePoints[1], FacePoints[2] };
+                    Point3 ClosestPoint0 = ClosestPointToTriangle(trianglePoints, point);
+                    CloserPointToFace_List.Add(ClosestPoint0);
+                }
+                else if (GSVertices.Count == 4)
+                {
+                    var trianglePoints1 = new Point3[] { FacePoints[0], FacePoints[1], FacePoints[2] };
+                    var trianglePoints2 = new Point3[] { FacePoints[1], FacePoints[2], FacePoints[3] };
+                    var trianglePoints3 = new Point3[] { FacePoints[0], FacePoints[1], FacePoints[3] };
+                    var trianglePoints4 = new Point3[] { FacePoints[0], FacePoints[2], FacePoints[3] };
+
+                    var ClosestPoint1 = ClosestPointToTriangle(trianglePoints1, point);
+                    var ClosestPoint2 = ClosestPointToTriangle(trianglePoints2, point);
+                    var ClosestPoint3 = ClosestPointToTriangle(trianglePoints3, point);
+                    var ClosestPoint4 = ClosestPointToTriangle(trianglePoints4, point);
+
+                    var TrianglesCP = new Point3[4] { ClosestPoint1, ClosestPoint2, ClosestPoint3, ClosestPoint4 };
+                    Point3 ClosestPoint = point.CloudClosestPoint(TrianglesCP);
+
+                    CloserPointToFace_List.Add(ClosestPoint);
+                }
+                else 
+                {
+                    //throw new Exception("Ngon detected");
+                    // Stellate method (from Ngon to triangles) 
```

Comment:
Is a "Stellate method" a name universally accepted? I'd rather just drop 2 lines of explanation. Something like 
_Triangulate the face using the centroid and blabla_